lib/rules/no-unused-class-component-methods.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @fileoverview Prevent declaring unused methods of component class
+ * @author Paweł Nowak
+ */
+
+'use strict';
+
+const Components = require('../util/Components');
+const astUtil = require('../util/ast');
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const internalMethods = [
+  'constructor',
+  'componentDidCatch',
+  'componentDidMount',
+  'componentDidUpdate',
+  'componentWillMount',
+  'componentWillReceiveProps',
+  'componentWillUnmount',
+  'componentWillUpdate',
+  'getSnapshotBeforeUpdate',
+  'render',
+  'shouldComponentUpdate',
+  'UNSAFE_componentWillMount',
+  'UNSAFE_componentWillReceiveProps',
+  'UNSAFE_componentWillUpdate'
+];
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent declaring unused methods of component class',
+      category: 'Best Practices',
+      recommended: false,
+      url: docsUrl('no-unused-class-component-methods')
+    },
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: false
+      }
+    ]
+  },
+
+  create: Components.detect((context, components, utils) => {
+    const isNotComponent = node => (
```

Comment:
shouldn't `Components.detect` give you a confidence value you can use here, without having to explicitly check component types?

Reply:
You are correct! I'll fix it.

Reply:
TBH I have some problem. `Components.detect` adds components to components list, so I can access it with `component.list()` yet still I would need to filter only components that I'm interested in. Wouldn't that be redundant?

Reply:
no, because that way the same logic used throughout the plugin is doing the initial filtering.